src/probe/probes-location-filter.ts

```diff
@@ -59,7 +59,7 @@ export class ProbesLocationFilter {
 	}
 
 	public filterByLocation (probes: Probe[], location: Location): Probe[] {
-		if (location.magic === 'world') {
+		if (location.magic?.toLowerCase() === 'world') {
```

Comment:
This would still throw if `magic` is undefined. Needs to be `location.magic?.toLowerCase?.()` to have any effect.

Reply:
`magic: undefined` is not possible from the app logic since we have `exactOptionalPropertyTypes: true`, and such case will break further anyway because we use constructions like `Object.keys(location).forEach()` and `Object.keys(location).includes('magic')`.

Reply:
Well then why is there the first `?.` if it's not possible?

Reply:
`{someOtherField: 'value'}` is possible `location.magic?.toLowerCase()` check is for that case. 
Explicit `{someOtherField: 'value', magic: undefined}` is not possible.

Reply:
Seems in any of the undefined cases expression will not get to `toLowerCase()`, so second `?` is not required.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#short-circuiting

Reply:
Oh indeed, I confused this with another case 👍